File: examples/nodejs/src/snippets/barcode/detect-barcodes.ts
import * as ScanbotSDK from "scanbotsdk";
import { printGenericDocument } from "../utils/utils";

export class DetectBarcodesSnippet {
  public static async run(image: ScanbotSDK.ImageRef): Promise<void> {
    var config = new ScanbotSDK.BarcodeScannerConfiguration();
    var formatConfig = new ScanbotSDK.BarcodeFormatCommonConfiguration();
    formatConfig.gs1Handling = "DECODE_FULL"
    
    config.barcodeFormatConfigurations = [formatConfig];
    config.returnBarcodeImage = true;
    // Configure other parameters as needed.

    // `await using` ensures both scanner and result are properly disposed
    // when the scope ends, as they hold unmanaged resources.
    await using scanner = await ScanbotSDK.BarcodeScanner.create(config);
    await using result = await scanner.run(image);
    const barcodes = result.barcodes ?? [];
    if (barcodes.length === 0) {
      console.log("No barcodes found.");
      return;
    }

    barcodes.forEach((barcode, idx) => {
      console.log(`Barcode #${idx + 1}:`);
      console.log(`  Format: ${barcode.format}`);
      console.log(`  Text:   ${barcode.text}`);

      printGenericDocument(barcode.extractedDocument);
    });
  }
}

File: examples/python/snippets/barcode/detect_barcodes.py
from scanbotsdk import *
from typing import List


def scan_barcode(image: ImageRef):
    configuration = BarcodeScannerConfiguration(
        return_barcode_image=True
    )
    configuration.barcode_format_configurations = [
        BarcodeFormatCommonConfiguration(
            gs_1_handling=Gs1Handling.DECODE_FULL,
        )
    ]

    scanner = BarcodeScanner(configuration=configuration)
    result: BarcodeScannerResult = scanner.run(image=image)

    print_barcodes(result.barcodes)


def print_barcodes(barcodes: List[BarcodeItem]):
    if not barcodes:
        print("No barcodes found")
        return

    print(f"Found {len(barcodes)} barcodes:")
    for i, barcode in enumerate(barcodes, start=1):
        print(f"Barcode #{i}:")
        print(f"\tFormat: {barcode.format.name}")
        print(f"\tText:   {barcode.text}")
